ext/console/02_console.js

```diff
@@ -1765,6 +1765,70 @@ function colorEquals(color1, color2) {
     color1?.[2] == color2?.[2];
 }
 
+let COLOR_SUPPORT_LEVEL = -1;
+let COLOR_SUPPORT_DUMB = 0;
+let COLOR_SUPPORT_256 = 1;
+let COLOR_SUPPORT_TRUE_COLOR = 2;
+
+const ANSI_FOREGROUND = "38";
+const ANSI_BACKGROUND = "48";
+
+/**
+ * Convert an RGB color to the color level that the terminal we're
+ * running in supports. We'll try to match requested color as close
+ * as possible to the colors we have available in case it lies outside
+ * the supported spectrum.
+ * @param {number} r red
+ * @param {number} g green
+ * @param {number} b blue
+ * @param {"38" | "48"} kind foreground or background color
+ * @returns {string}
+ */
+function rgbToAnsi(r, g, b, kind) {
+  // Lazily check for color support
+  if (COLOR_SUPPORT_LEVEL === -1) {
+    const COLORTERM = Deno.env.get("COLORTERM");
+    COLOR_SUPPORT_LEVEL = COLORTERM === "truecolor" || COLORTERM === "24bit"
+      ? COLOR_SUPPORT_TRUE_COLOR
+      : Deno.env.get("TERM").endsWith("-256color")
```

Comment:
These will prompt for env permissions on first access. Maybe we could check these env vars in Rust and forward this info during runtime bootstrapping, similarly how we do it for `NO_COLOR` variable?

https://github.com/denoland/deno/blob/ae1ba2af3c0dc45de074285f0e0a8440cf2895ec/runtime/js/99_main.js#L315

Reply:
Ohh right that's a good point! Passing them from rust makes much more sense. Will update the PR.

Reply:
Sounds good!

Reply:
Moved the color level detection code over to rust. Thanks again for the idea!